### PR TITLE
Update CodeNarc to 3.4.0

### PR DESCRIPTION
**Pull request by nre-ableton**
- Update CodeNarc to 3.4.0
- Version 0.14.6

### CODE REVIEW

**Comment by mst-ableton** [comment]
I think something went wrong here... you have a merge commit instead of having 7fa261ff90a0d03ccc653fc7d904b0d9982c918f on its own. Other than that it looks good.

File: tests/resources/failure.badgroovy [hunk 1]
```diff
@@ -0,0 +1 @@
+def foo = 'hello'
```
Comment (mst-ableton) [inline tests/resources/failure.badgroovy:1]: Could you put a comment here saying that this should trigger ` VariableTypeRequired`?

Comment (nre-ableton) [reply tests/resources/failure.badgroovy]: See [8dc54f7](https://github.com/Ableton/groovylint/pull/411/commits/8dc54f7be796975f00c66ecbfeac6060e9c45486).

**Review by mst-ableton — APPROVED** [review]
lgtm, thanks!

**Pull request merged**